expression/builtin_other.go

```diff
@@ -1108,7 +1108,11 @@ func (c *getTimeVarFunctionClass) getFunction(ctx sessionctx.Context, args []Exp
 	if err != nil {
 		return nil, err
 	}
-	bf.tp.Flen = c.tp.Flen
+	fsp := args[0].GetType().Flen - mysql.MaxDatetimeWidthNoFsp
+	if fsp > 0 {
+		fsp--
+	}
+	bf.setDecimalAndFlenForDatetime(fsp)
```

Comment:
How do the `Decimal` and `Flen` affect the behavior?
```sql
mysql> set @a = cast('2000-01-01' as date);
mysql> select @a;
Field   1:  `@a`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       DATETIME
Collation:  binary (63)
Length:     0
Max_length: 10
Decimals:   236
Flags:      BINARY 


+------------+
| @a         |
+------------+
| 2000-01-01 |
+------------+
```

We can see the decimal is `236`, which is the unsigned version for `-20`. Is it expected?

Reply:
Fixed